### PR TITLE
migrate to new version 2.x more food of methods

### PR DESCRIPTION
**Pull request by waadmawlood**
## Features and Improvements
- Implemented `HasOneMetadata` trait for single metadata record per model
  - CRUD operations for metadata (create, read, update, delete)
  - Methods for checking metadata existence and content
  - Support for multiple data types (string, bool, null, int, float, array)
  - Collection and array return type support

- Enhanced `HasManyMetadata` trait for multiple metadata records per model
  - Support for creating multiple metadata records
  - Configurable metadata ID appending with `metadataNameIdEnabled` and `metadataNameId`
  - Advanced search and filtering capabilities
  - Bulk operations support

## Key Methods Added/Enhanced
- `createMetadata()` - Create new metadata record(s)
- `getMetadata()` - Retrieve metadata as array
- `getMetadataCollection()` - Retrieve metadata as collection
- `updateMetadata()` - Update existing metadata
- `deleteMetadata()` - Remove metadata
- `hasMetadata()`, `hasFilledMetadata()` - Check metadata existence
- `addKeysMetadata()`, `forgetKeysMetadata()` - Manage specific metadata keys

## Testing
- Comprehensive test coverage for both traits
- Test cases for all data types and edge cases

## Documentation
- Updated documentation with usage examples
- Added configuration details for metadata ID appending
- Improved method documentation and type hints
[Documentation](https://waad-mawlood.gitbook.io/model-metadata)

### CODE REVIEW

File: src/Providers/MetadataServiceProvider.php [hunk 1]
```diff
@@ -3,23 +3,23 @@
...
 
 use Illuminate\Support\ServiceProvider;
+use Waad\Metadata\Helpers\Helper;
 
 class MetadataServiceProvider extends ServiceProvider
 {
     public function register()
     {
-        //
+        $this->app->singleton(Helper::class);
```
Comment (waadmawlood) [inline src/Providers/MetadataServiceProvider.php:12]: this is support service container of laravel

**Review by waadmawlood — COMMENTED** [review]
done
